src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -450,112 +449,116 @@ protected virtual string GenerateUniqueId()
             }
         }
 
-        private void RenderUnboundAttribute(string attributeName, Chunk attributeValueChunk)
+        private void RenderUnboundAttribute(TagHelperAttributeChunk attribute)
         {
             // Render children to provide IntelliSense at design time. No need for the execution context logic, it's
             // a runtime feature.
             if (_designTimeMode)
             {
-                if (attributeValueChunk != null)
+                if (attribute.Value != null)
                 {
-                    _bodyVisitor.Accept(attributeValueChunk);
+                    _bodyVisitor.Accept(attribute.Value);
                 }
 
                 return;
             }
 
-            // If we have a minimized attribute there is no value
```

Comment:
Please restore the order and condition priority of the old code. Very hard to see what's changing here or why the new order is any better than the old. If the new order is better, it's not related to tracking quotes. So, make the change in a separate PR.